### ISSUE TITLE
Refactor resetting fields

### ISSUE THREAD

**Issue by knjk04** [Project-Books/book-project#415]
**Context**

To reset fields, we are currently iterating over all fields and clearing them manually. This is both verbose and somewhat of a code smell as we are using raw types.

```java
    public static void clearComponentFields(HasValue... components) {
        for (HasValue component : components) {
            if (component != null && !component.isEmpty()) {
                component.clear();
            }
        }
    }
```

**What we want**

A more elegant solution that does not use raw types. One possible approach:

```java
    public static void clearComponentFields(Binder<Book> binder) {
        binder.getFields().forEach(HasValue::clear);
    }
```

Something like this (or better) but more generic would be nice (so that we can use it for other forms).

**Comment by guctum** [comment]
@knjk04 hello, I can take a look at this if needed! Is there a specific class to look at to make these changes in?

**Comment by knjk04** [comment]
Thanks, @guctum! I'll assign you to the issue

This is in `ui/components/utils/ComponentUtil.java`

**Comment by guctum** [comment]
@knjk04 quick question: in BookForm.java, following your proposed idea of converting the signature to Binder<> instead of the HasValue produces the following issue:
![Screen Shot 2020-10-04 at 5 12 20 PM](https://user-images.githubusercontent.com/22298395/95027217-dd0c5000-0664-11eb-882e-0aae0c73372b.png)
Are there better options that can be used for components? Not being very familiar with this codebase, I'm not sure what changes I can make to resolve this without running the risk of breaking more.

**Comment by knjk04** [comment]
@guctum To fix this error, you would need to pass in `binder` (which is of type `Binder`). 

If you need further help, please reach out to us on our [Slack](https://join.slack.com/t/teambookproject/shared_invite/zt-h6vx3n8l-KN8QnO50r7QWZHgHFnFdPw) workspace and we'll be happy to help

**Comment by guctum** [comment]
@knjk04 thanks for the patience and advice with this! Life got super busy, so I just got the chance to actually read and execute. PR has been opened, please let me know if there's anything further you need from me with this. Awesome project to work on here!! Gonna go look for more hacktoberfest issues with this project. With this said, can I get a new invite link to Slack? If I'm going to pick up on more issues, it might be beneficial to have other people I can reach out to with questions :)

Unrelated question to this issue, what made you choose to use Vaadin?

**Comment by knjk04** [comment]
@guctum 

> thanks for the patience and advice with this! Life got super busy, so I just got the chance to actually read and execute. PR has been opened, please let me know if there's anything further you need from me with this. 

My pleasure, I'm glad I helped.

> Awesome project to work on here!! Gonna go look for more hacktoberfest issues with this project. 

Thank you for your kind words!

> With this said, can I get a new invite link to Slack? If I'm going to pick up on more issues, it might be beneficial to have other people I can reach out to with questions :)

Sure, here's a [Slack invite](https://join.slack.com/t/teambookproject/shared_invite/zt-i7lept44-rgJ9yB0A2vedJTLyyfkjKQ)

> Unrelated question to this issue, what made you choose to use Vaadin?

I like Java, and I don't really know frontend languages and frameworks. Vaadin's a well designed and documented library. 

With that said, I was aware that it may not be scalable, so from early in, I included in our roadmap the possibility of switching over to Angular. Vaadin was great for quickly making a prototype, so I have no regrets. It also integrates nicely with Spring.

We're now at the point where we're planning on migrating to React (we decided against Angular in the end).

If you have any more questions, or any suggestions or feedback, feel free to ask/let us know! However, let's take this over to Slack as it's better for an open-ended discussion that's not related to this issue :)
